src/lib/sitemap/helpers.ts

```diff
@@ -0,0 +1,36 @@
+import { SitemapElement, SitemapFactoryInput, SitemapPriority } from './types'
+
+/**
+ *
+ * @param slug
+ * @returns 1 - 0.7 depending on slug depth in relation to a product's base path
+ *
+ * See XML Tag Definitions: https://www.sitemaps.org/protocol.html
+ */
+export const determinePriority = (slug: string): SitemapPriority => {
+	const slugDepth = slug.split('/').length
+	switch (slugDepth) {
+		case 1:
+			return 1
+		case 2:
+			return 0.9
+		case 3:
+			return 0.8
+		default:
+			return 0.7
+	}
+}
```

Comment:
❓ I'm curious on the reasoning behind this logic. Is there some resource that recommends this approach, or what's the thought here?

Reply:
Here is a decent [explanation of sitemap priority](https://www.contentpowered.com/blog/xml-sitemap-priority-changefreq/)

It feels kind of wrong to just default all urls to 0.7, so I though I'd do some simple prioritization logic. None of the current paths have a priority of 1 yet, and most are 0.7.

`/terraform/language` => 0.9
`/terraform/language/attr-as-blocks` => 0.8

Reply:
@nayres Makes sense!